client/src/main/java/hudson/plugins/swarm/SwarmClient.java

```diff
@@ -368,6 +395,53 @@ private static String getChildElementString(Element parent, String tagName) {
         return null;
     }
 
+    /**
+     * Returns a hash that should be consistent for any individual swarm client (as long as it has a persistent IP)
+     * and should be unique to that client.
+     *
+     * @param remoteFsRoot the file system root should be part of the hash (to support multiple swarm clients from 
+     *                     the same machine)
+     * @return our best effort at a consistent hash
+     */
+    public static String hash(File remoteFsRoot) {
+        StringBuilder buf = new StringBuilder();
+        try {
+            buf.append(remoteFsRoot.getCanonicalPath()).append('\n');
+        } catch (IOException e) {
+            buf.append(remoteFsRoot.getAbsolutePath()).append('\n');
+        }
+        try {
+            for (NetworkInterface ni : Collections.list(NetworkInterface.getNetworkInterfaces())) {
+                for (InetAddress ia : Collections.list(ni.getInetAddresses())) {
+                    if (ia instanceof Inet4Address) {
+                        buf.append(ia.getHostAddress()).append('\n');
+                    }
+                }
+                byte[] hardwareAddress = ni.getHardwareAddress();
```

Comment:
I do not think that Jenkins works on IPv6.

It'll most likely be dhcp for its v4 address anyway

On Friday, April 24, 2015, James Nord <notifications@github.com> wrote:

> In client/src/main/java/hudson/plugins/swarm/SwarmClient.java
> <https://github.com/jenkinsci/swarm-plugin/pull/26#discussion_r29063667>:
>
> > +     */
> > +    public static String hash(File remoteFsRoot) {
> > +        StringBuilder buf = new StringBuilder();
> > +        try {
> > +            buf.append(remoteFsRoot.getCanonicalPath()).append('\n');
> > +        } catch (IOException e) {
> > +            buf.append(remoteFsRoot.getAbsolutePath()).append('\n');
> > +        }
> > +        try {
> > +            for (NetworkInterface ni : Collections.list(NetworkInterface.getNetworkInterfaces())) {
> > +                for (InetAddress ia : Collections.list(ni.getInetAddresses())) {
> > +                    if (ia instanceof Inet4Address) {
> > +                        buf.append(ia.getHostAddress()).append('\n');
> > +                    }
> > +                }
> > +                byte[] hardwareAddress = ni.getHardwareAddress();
>
> so for IPv6 hosts when a machine is cloned this is going to break?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkinsci/swarm-plugin/pull/26/files#r29063667>.
>


-- 
Sent from my phone